src/mds/MDSRank.h

```diff
@@ -228,7 +392,7 @@ class MDSRank {
     }
 
     void handle_write_error(int err);
-    void handle_write_error_with_lock(int err);
+    void handle_write_error_unlocked(int err);
```

Comment:
Can you rename functions as a separate commit? I happened to notice it while we were talking and then had to scramble to figure out why you were changing the lock state, only to realize you had renamed the function (presumably you found it misleading?).

Reply:
Yes please -- it makes the correctness checks way easier if renames are independent of anything else. :)